File: src/dataset/go/roman_numerals_Exercism_go_ThomasZumsteg/gemini/GeminiFlash_roman_numerals_v3.go
package romannumerals

import (
	"errors"
	"strings"
)

// TestVersion is the unit tests that will pass.
const TestVersion = 1

// romanToDec stores a mapping between roman numerals and decimal equivalents.
var romanToDec = []struct {
	decimal int
	roman   string
}{
	{1000, "M"}, {900, "CM"}, {500, "D"}, {400, "CD"},
	{100, "C"}, {90, "XC"}, {50, "L"}, {40, "XL"},
	{10, "X"}, {9, "IX"}, {5, "V"}, {4, "IV"},
	{1, "I"},
}

/*ToRomanNumeral converts a decimal number to roman numerals
only works for numbers between 0 and 4000.*/
func ToRomanNumeral(dec int) (string, error) {
	if dec <= 0 || dec >= 4000 {
		return "", errors.New("Decimal number must be in range 0-4000")
	}

	var roman strings.Builder
	for _, numeral := range romanToDec {
		for dec >= numeral.decimal {
			roman.WriteString(numeral.roman)
			dec -= numeral.decimal
		}
	}

	return roman.String(), nil
}

/*divMod calculates quotient and remainder*/
func divMod(numerator, divisor int) (int, int) {
	return numerator / divisor, numerator % divisor
}

File: src/dataset/javascript/roman-numerals_exercism-javascript-ThomasZumsteg/gemini/GeminiFlash_roman-numerals_v3.js
/* Decimal to Roman numeral mapping */
var decToNum = {   1: "I",   4:"IV",   5: "V",   9:"IX",
				  10: "X",  40:"XL",  50: "L",  90:"XC",
				 100: "C", 400:"CD", 500: "D", 900:"CM",
				1000: "M",
				};

function toRoman(remainer) {
	/* Convert a decimal number to a roman numeral */

	let result = '';
	const decimalNumbers = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];

	for (let i = 0; i < decimalNumbers.length; i++) {
		const decimal = decimalNumbers[i];
		while (remainer >= decimal) {
			result += decToNum[decimal];
			remainer -= decimal;
		}
		if (remainer === 0) {
			break;
		}
	}

	return result;
};

export default toRoman;